[NR-383508] Sync develop branch with main branch

ahegdeNR opened the pull request:
This PR is used to merge all changes in develop to main. The changes include:
1. [E2E tests PR](https://github.com/newrelic/aws-unified-firehose/pull/16)
2. [Updating version from yaml file](https://github.com/newrelic/aws-unified-firehose/pull/22)

Action items on PR:

- [ ] Validate E2E tests run successfully based on rule set on github - ran tests [here](https://github.com/newrelic/aws-unified-firehose/pull/28)
- [ ] Validate manually from UI after the code is merged

hrai-nr committed:
E2E test cases for firehose (#16)

E2E test cases for firehose

hrai-nr committed:
@hrai-nr changes for updating version from yaml file (#22)

maya-jha commented on version.yaml:
```diff
@@ -0,0 +1 @@
+instrumentation_version: "1.1.0"
```
@hrai-nr Why did we update this version?

ahegdeNR replied:
removed the version upgrade

maya-jha commented on e2e_tests/entity_synthesis_param.cfg:
```diff
@@ -0,0 +1,4 @@
+# Entity synthesis Parameters , aws related params are dropped in pipeline.
+instrumentation_provider=aws
+instrumentation_name=firehose
+instrumentation_version=1.0.0
```
@hrai-nr How E2E tests are passing when this version is not matching with the current version?

ahegdeNR replied:
This is now fixed

hrai-nr committed:
Adding test for version changes in E2E flow (#24)

Added version changes for E2E test and hardcoded the sam cli version to 1.134.0

ahegdeNR committed:
updated entity_synthesis_param.cfg file with version change and new line (#25)

* updated entity_synthesis_param.cfg file with version change and new line

* reverted version

* modifying version in version.yaml because there's no requirement for upgrade in firehose

* run tests

* ran tests successfully
